examples/widgets/scale/lv_example_scale_10.c

```diff
@@ -0,0 +1,175 @@
+static section_styles_t zone3_styles;
+static section_styles_t zone4_styles;
+static section_styles_t zone5_styles;
+
+static lv_color_t get_hr_zone_color(int32_t hr)
+{
+    if(hr < 117) return lv_palette_main(LV_PALETTE_GREY);      // Zone 1
+    else if(hr < 135) return lv_palette_main(LV_PALETTE_BLUE); // Zone 2
+    else if(hr < 158) return lv_palette_main(LV_PALETTE_GREEN);// Zone 3
+    else if(hr < 176) return lv_palette_main(LV_PALETTE_ORANGE);// Zone 4
+    else return lv_palette_main(LV_PALETTE_RED);               // Zone 5
+}
+
+static void hr_anim_timer_cb(lv_timer_t * timer)
+{
+    LV_UNUSED(timer);
+
+    hr_value += hr_step;
+
+    if(hr_value >= 195) {
+        hr_value = 195;
+        hr_step = -1;
+    }
+    else if(hr_value <= 98) {
+        hr_value = 98;
+        hr_step = 1;
+    }
+
+    /* Update needle */
+    lv_scale_set_line_needle_value(scale, needle_line, -8, hr_value);
+
+    /* Update HR text */
+    lv_label_set_text_fmt(hr_value_label, "%d", hr_value);
+
+    /* Update text color based on zone */
+    lv_color_t zone_color = get_hr_zone_color(hr_value);
+    lv_obj_set_style_text_color(hr_value_label, zone_color, 0);
+    lv_obj_set_style_text_color(bpm_label, zone_color, 0);
+}
+
+static void init_section_styles(section_styles_t * styles, lv_color_t color)
+{
+    lv_style_init(&styles->items);
+    lv_style_set_line_color(&styles->items, color);
+    lv_style_set_line_width(&styles->items, 0);
+
+    lv_style_init(&styles->indicator);
+    lv_style_set_line_color(&styles->indicator, color);
+    lv_style_set_line_width(&styles->indicator, 0);
+
+    lv_style_init(&styles->main);
+    lv_style_set_arc_color(&styles->main, color);
+    lv_style_set_arc_width(&styles->main, 20);
+}
+
+static void add_section(lv_obj_t * target_scale,
+                        int32_t from,
+                        int32_t to,
+                        const section_styles_t * styles)
+{
+    lv_scale_section_t * sec = lv_scale_add_section(target_scale);
+    lv_scale_set_section_range(target_scale, sec, from, to);
+    lv_scale_set_section_style_items(target_scale, sec, &styles->items);
+    lv_scale_set_section_style_indicator(target_scale, sec, &styles->indicator);
+    lv_scale_set_section_style_main(target_scale, sec, &styles->main);
+}
+
+void lv_example_scale_10(void)
+{
+    scale = lv_scale_create(lv_screen_active());
+    lv_obj_center(scale);
+    lv_obj_set_size(scale, 200, 200);
+
+    lv_scale_set_mode(scale, LV_SCALE_MODE_ROUND_INNER);
+    lv_scale_set_range(scale, 98, 195);
+    lv_scale_set_total_tick_count(scale, 15);
+    lv_scale_set_major_tick_every(scale, 3);
+    lv_scale_set_angle_range(scale, 280);
+    lv_scale_set_rotation(scale, 130);
+    lv_scale_set_label_show(scale, false);
+
+    lv_obj_set_style_length(scale, 6, LV_PART_ITEMS);
+    lv_obj_set_style_length(scale, 10, LV_PART_INDICATOR);
+    lv_obj_set_style_arc_width(scale, 0, LV_PART_MAIN);
+
+    // Zone 1: (Grey)
```

Comment:
Please use this comment style
```suggestion
    /* Zone 1: (Grey) */
```